### PR TITLE
Added tests for useStickyHeader hook

### LINKED ISSUE #89: (SP: ) Create tests for useStickyHeader.ts

### PR DESCRIPTION
**Pull request by heshvab**
- Added tests for useStickyHeader hook

Unit tests coverage
image_2024-06-30_16-50-05

Mutation tests coverage
image

### CODE REVIEW

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { renderHook } from "@testing-library/react";
+import useStickyHeader from "./useStickyHeader";
```
Comment (D1nviz) [inline code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: use alias here

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    initialHeaderElement = document.createElement("div");
+    initialScrollHandleElement = document.createElement("div");
+
+    observerOptions = undefined;
+    observerCallback = () => {};
+
+    jest.clearAllMocks();
+  });
+
+  it("Should not call observe function or return ref if there is no at least one ref assigned to html element", async () => {
```
Comment (D1nviz) [inline code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: can we avoid using `async` for each case?

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    const { result } = renderUseStickyHeader({
+      headerElement: null
+    });
+
+    expect(result.current.headerRef.current).toBeNull();
+
+    expect(mockedObserveFn).not.toHaveBeenCalled();
+  });
+
+  it("Should call observe and unobserve if refs are asigned", async () => {
```
Comment (D1nviz) [inline code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: also, use `test` instead of `it`

**Review by mhevyk — APPROVED** [review]
Amazing job! You are very strong developer!

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts (hunk 1 above)
Comment (heorhiiss) [reply code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: Thanks, fixed

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts (hunk 2 above)
Comment (heorhiiss) [reply code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: Thanks for noticing, I mistakenly added it. FIxed

File: code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts (hunk 3 above)
Comment (heorhiiss) [reply code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts]: Thanks, I used test there and in other places

**Review by D1nviz — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**